pallets/nfts/src/lib.rs

```diff
@@ -1371,6 +1526,55 @@ pub mod pallet {
 			Self::do_clear_all_transfer_approvals(maybe_check_origin, collection, item)
 		}
 
+		/// Cancel collection approvals, up to a specified limit.
+		///
+		/// Origin must be Signed.
+		///
+		/// Arguments:
+		/// - `collection`: The collection whose approvals will be cleared.
+		/// - `limit`: The amount of collection approvals that will be cleared.
+		///
+		/// Emits `AllApprovalsCancelled` on success.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(43)]
+		#[pallet::weight(T::WeightInfo::clear_collection_approvals_limit(*limit))]
+		pub fn clear_collection_approvals_limit(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			limit: u32,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			Self::do_clear_collection_approvals_limit(origin, collection, limit)?;
+			Ok(())
+		}
+
+		/// Force-cancel collection approvals granted by `owner` account, up to a specified limit.
+		///
+		/// Origin must be `ForceOrigin`.
+		///
+		/// Arguments:
+		/// - `owner`: The account clearing all collection approvals.
+		/// - `collection`: The collection whose approvals will be cleared.
+		/// - `limit`: The amount of collection approvals that will be cleared.
+		///
+		/// Emits `AllApprovalsCancelled` on success.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(44)]
+		#[pallet::weight(T::WeightInfo::force_clear_collection_approvals_limit(*limit))]
+		pub fn force_clear_collection_approvals_limit(
```

Comment:
```suggestion
		pub fn force_clear_collection_approvals(
```
I dont know that we need the _limit suffix as there is nothing to disambiguate and it therefore just adds noise.

Reply:
@Daanvdplas This is recommended by you so do you agree with changing it?

Reply:
Never recommend it for the name (or intended to), I meant the parameter.